server/src/handlers/influxdb.rs

```diff
@@ -71,6 +77,303 @@ impl From<Bytes> for WriteRequest {
 
 pub type WriteResponse = ();
 
+/// Influxql response organized in the same way with influxdb
+#[derive(Debug, Serialize)]
+pub struct InfluxqlResponse {
+    pub results: Vec<InfluxqlResult>,
+}
+
+#[derive(Debug)]
+pub struct InfluxqlResult {
+    statement_id: u32,
+    series: Option<Vec<OneSeries>>,
+}
+
+impl Serialize for InfluxqlResult {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut influxql_result = serializer.serialize_map(Some(4))?;
+        influxql_result.serialize_entry("statement_id", &self.statement_id)?;
+        if let Some(series) = &self.series {
+            influxql_result.serialize_entry("series", series)?;
+        }
+
+        influxql_result.end()
+    }
+}
+
+#[derive(Debug)]
+struct OneSeries {
+    name: String,
+    tags: Option<Tags>,
+    columns: Vec<String>,
+    values: Vec<Vec<Datum>>,
+}
+
+impl Serialize for OneSeries {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut one_series = serializer.serialize_map(Some(4))?;
+        one_series.serialize_entry("name", &self.name)?;
+        if let Some(tags) = &self.tags {
+            one_series.serialize_entry("tags", &tags)?;
+        }
+        one_series.serialize_entry("columns", &self.columns)?;
+        one_series.serialize_entry("values", &self.values)?;
+
+        one_series.end()
+    }
+}
+
+#[derive(Debug)]
+struct Tags(Vec<(String, String)>);
+
+impl Serialize for Tags {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut tags = serializer.serialize_map(Some(self.0.len()))?;
+
+        for (tagk, tagv) in &self.0 {
+            tags.serialize_entry(tagk, tagv)?;
+        }
+
+        tags.end()
+    }
+}
+
+/// [InfluxqlResult] builder
+#[derive(Default)]
+pub struct InfluxqlResultBuilder {
+    /// Id of the related influxql
+    statement_id: u32,
+
+    /// Schema of influxql query result
+    ///
+    /// Its format is like:
+    /// measurement |
+    /// tag_1..tag_n(columns in `group by`) |
+    /// time |
+    /// column_1..column_n(column in `projection` but not in `group by`)
+    column_schemas: Vec<ColumnSchema>,
+
+    /// Tags part in schema
+    group_by_tag_col_idxs: Vec<usize>,
+
+    /// Columns part in schema(include `time`)
+    value_col_idxs: Vec<usize>,
+
+    /// Mapping group key(`measurement` + `tag values`) to column values,
+    ///
+    /// NOTE: because tag keys in `group by` clause are same in each sub result,
+    /// we just use the `measurement` + `tag values` to distinguish them.
+    group_key_to_idx: HashMap<GroupKey, usize>,
+
+    /// Column values
+    value_groups: Vec<Vec<Vec<Datum>>>,
```

Comment:
Add comments explain meaning of different level of Vec, or you can add some type alias to eliminate ambiguities